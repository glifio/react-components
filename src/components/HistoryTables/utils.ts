import { FilecoinNumber, BigNumber } from '@glif/filecoin-number'
import { MessageConfirmed } from '../../generated/graphql'
import makeFriendlyBalance from '../../utils/makeFriendlyBalance'

export function attoFilToFil(amount: string | number | BigNumber): string {
  return new FilecoinNumber(amount, 'attofil').toFil() + ' FIL'
}

export function getTotalCost(
  message: Pick<
    MessageConfirmed,
    'baseFeeBurn' | 'overEstimationBurn' | 'minerTip'
  >
): string {
  const bnBaseFeeBurn = new BigNumber(message.baseFeeBurn)
  const bnOverEstimationBurn = new BigNumber(message.overEstimationBurn)
  const bnMinerTip = new BigNumber(message.minerTip)
  const totalCost = bnBaseFeeBurn.plus(bnOverEstimationBurn).plus(bnMinerTip)
  return new FilecoinNumber(totalCost, 'attofil').toFil() + ' FIL'
}

export function getTotalCostShort(
  message: Pick<
    MessageConfirmed,
    'baseFeeBurn' | 'overEstimationBurn' | 'minerTip'
  >
): string {
  const bnBaseFeeBurn = new BigNumber(message.baseFeeBurn)
  const bnOverEstimationBurn = new BigNumber(message.overEstimationBurn)
  const bnMinerTip = new BigNumber(message.minerTip)
  const totalCost = bnBaseFeeBurn.plus(bnOverEstimationBurn).plus(bnMinerTip)
  const totalFil = new FilecoinNumber(totalCost, 'attofil')
  return makeFriendlyBalance(totalFil) + ' FIL'
}

export function getGasPercentage(
  message: Pick<MessageConfirmed, 'gasLimit' | 'gasUsed'>
): string {
  const gasLimit = new BigNumber(message.gasLimit)
  const gasUsed = new BigNumber(message.gasUsed)
  return gasUsed.dividedBy(gasLimit).times(100).toFixed(1) + '%'
}

export function formatNumber(input: string | number | BigNumber): string {
  const bigNr = input instanceof BigNumber ? input : new BigNumber(input)
  return bigNr.toFormat()
}
