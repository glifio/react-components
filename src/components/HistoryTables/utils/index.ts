import { FilecoinNumber, BigNumber } from '@glif/filecoin-number'
import { Message, MessagePending } from '../../../generated/graphql'

export function attoFilToFil(amount: string | number | BigNumber): string {
  return new FilecoinNumber(amount, 'attofil').toFil() + ' FIL'
}

export function getGasPercentage(
  message: Pick<Message, 'gasLimit' | 'gasCost'> | MessagePending,
  pending: boolean
): string {
  if (pending) return ''
  message = message as Message
  const gasLimit = new BigNumber(message.gasLimit)
  const gasUsed = new BigNumber(message.gasCost.gasUsed)
  return gasUsed.dividedBy(gasLimit).times(100).toFixed(1) + '%'
}

export function formatNumber(input: string | number | BigNumber): string {
  const bigNr = input instanceof BigNumber ? input : new BigNumber(input)
  return bigNr.toFormat()
}

export function removeMessageDups(
  existing: { __ref: any }[],
  incoming: { __ref: any }[],
  args: { offset: number; limit: number }
) {
  const head = [...existing]
  const tail = head.splice(args.offset * args.limit)
  const merged = [].concat(head, incoming, tail)

  const found = new Set([])
  return [...merged].filter(ele => {
    if (!found.has(ele.__ref)) {
      found.add(ele.__ref)
      return true
    }

    return false
  })
}
