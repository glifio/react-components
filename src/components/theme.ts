export const baseColors = {
  mono: {
    black: '#000',
    nearblack: '#262626',
    darkgray: '#666666',
    lightgray: '#C4C4C4',
    silver: '#999999',
    white: '#ffffff',
    transparent: 'transparent'
  },
  blue: {
    lightest: '#F6F8FE',
    lighter: '#EFF3FD',
    light: '#E4EBFC',
    mid: '#d1ddfa',
    mid2: '#bacbf7',
    muted: '#e8ecf7',
    muted700: '#d9e0f2'
  },
  green: {
    primary: '#1AD08F',
    light: '#D2F5ED',
    muted: '#85e0cb',
    dark: '#007056',
    darker: '#08442F'
  },
  red: {
    light: '#FC6D6F',
    dark: '#660000',
    base: '#ff4242'
  },
  orange: {
    light: '#FF9E80',
    dark: '#661800',
    mid: '#FFA826'
  },
  yellow: {
    light: '#FFDC99',
    deep: '#FCA703',
    mid: '#F7C103'
  },
  purple: {
    light: '#E0D7FE',
    mid: '#5E26FF',
    deep: '#290a85'
  }
}

// The core color object is the only non-semantic object here. This is because these colors are used so widely that it would be highly redundant to replicate these color values repeatedly within this "colors" object to style the text/background of every single Component the app renders.
const core = {
  primary: baseColors.purple.mid,
  secondary: baseColors.purple.light,
  tertiary: baseColors.purple.deep,
  black: baseColors.mono.black,
  nearblack: baseColors.mono.nearblack,
  darkgray: baseColors.mono.darkgray,
  lightgray: baseColors.mono.lightgray,
  silver: baseColors.mono.silver,
  white: baseColors.mono.white,
  transparent: baseColors.mono.transparent,
  yellow: baseColors.yellow.deep
}

const colors = {
  core,
  buttons: {
    primary: {
      background: baseColors.green.primary,
      borderColor: baseColors.green.primary,
      color: baseColors.green.darker
    },
    secondary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.mono.nearblack,
      color: baseColors.mono.nearblack
    },
    tertiary: {
      background: baseColors.mono.transparent,
      borderColor: baseColors.purple.light,
      color: baseColors.purple.light
    },
    'tertiary-selected': {
      background: baseColors.purple.mid,
      borderColor: baseColors.purple.mid,
      color: baseColors.purple.light
    }
  },
  background: {
    app: baseColors.blue.lightest,
    screen: baseColors.blue.lighter,
    text: baseColors.blue.light,
    messageHistory: baseColors.blue.muted
  },
  card: {
    account: {
      background: core.primary,
      color: core.secondary
    },
    balance: {
      background: baseColors.mono.transparent,
      color: core.nearblack
    },
    confirmation: {
      background: baseColors.green.light,
      foreground: baseColors.green.dark
    },
    error: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    }
  },
  input: {
    background: {
      base: baseColors.blue.mid,
      active: baseColors.blue.mid2,
      valid: baseColors.green.muted,
      invalid: baseColors.red.light
    },
    border: core.silver
  },
  status: {
    success: {
      background: baseColors.green.primary,
      foreground: baseColors.green.darker
    },
    pending: {
      background: baseColors.yellow.deep,
      foreground: baseColors.mono.darkgray
    },
    warning: {
      background: baseColors.orange.light,
      foreground: baseColors.orange.dark
    },
    fail: {
      background: baseColors.red.light,
      foreground: baseColors.red.dark
    },
    inactive: baseColors.mono.lightgray
  },
  // gonna start slowly migrating colors from their inheritance vars above
  blue: {
    muted700: '#d9e0f2'
  },
  gray: {
    light: '#C4C4C4',
    medium: '#949598',
    dark: '#666666'
  },
  green: {
    primary: '#1AD08F',
    light: '#D2F5ED',
    muted: '#85e0cb',
    medium: '#007B39',
    dark: '#007056',
    darker: '#08442F'
  },
  yellow: {
    light: '#FFDC99',
    medium: '#C59900',
    deep: '#FCA703',
    mid: '#F7C103'
  },
  purple: {
    light: '#E0D7FE',
    medium: '#5E26FF',
    deep: '#290a85'
  }
}

const theme = {
  colors,
  fontSizes: [
    '0rem',
    '1rem',
    '1.125rem',
    '1.25rem',
    '1.5rem',
    '2rem',
    '3rem',
    '5rem'
  ],
  fontWeights: [0, 400, 600, 900],
  letterSpacings: [0, 1, 2, 4, 8],
  lineHeights: {
    solid: 1,
    title: 1.2,
    copy: 1.4
  },
  textStyles: {
    header: {
      fontSize: 'calc(48px + (48 - 32) * (100vw - 360px) / (1440 - 360))',
      fontWeight: 400,
      margin: 0,
      lineHeight: 'title',
      fontFamily: 'RT-Alias-Grotesk'
    },
    bigTitle: {
      fontSize: 6,
      fontWeight: 700,
      margin: 0,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Medium'
    },
    title: {
      fontSize: 4,
      fontWeight: 400,
      margin: 0,
      lineHeight: 'title',
      fontFamily: 'RT-Alias-Grotesk'
    },
    text: {
      fontSize: 2,
      fontColor: core.nearblack,
      fontWeight: 400,
      lineHeight: 'copy',
      fontFamily: 'RT-Alias-Grotesk'
    },
    label: {
      fontSize: 2,
      fontColor: core.darkgray,
      fontWeight: 400,
      lineHeight: 'solid',
      fontFamily: 'RT-Alias-Grotesk'
    },
    num: {
      m: {
        fontSize: 4,
        fontWeight: 400,
        margin: 0,
        lineHeight: 'title',
        fontFamily: 'RT-Alias-Grotesk'
      },
      l: {
        fontSize: 5,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'title',
        fontFamily: 'RT-Alias-Grotesk'
      },
      xl: {
        fontSize: 6,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'solid',
        fontFamily: 'RT-Alias-Grotesk'
      },
      xxl: {
        fontSize: 7,
        fontWeight: 700,
        margin: 0,
        lineHeight: 'solid',
        fontFamily: 'RT-Alias-Grotesk'
      }
    }
  },
  fonts: {
    AliasMedium:
      '"RT-Alias-Medium", "system-ui", "Segoe UI", "Roboto", Helvetica',
    AliasGrotesk:
      '"RT-Alias-Grotesk", "system-ui", "Segoe UI", "Roboto", Helvetica',
    sansSerif:
      '"system-ui", "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";',
    mono: '"system-ui", "Segoe UI", Roboto Mono, Helvetica, Arial, monospace, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";'
  },
  space: [0, 4, 8, 16, 24, 32, 48, 64, 128, 256],
  sizes: [
    0, 4, 8, 16, 24, 32, 48, 64, 80, 160, 240, 300, 480, 560, 640, 720, 800,
    960, 1024, 1200, 1440
  ],
  radii: ['0', '1px', '4px', '8px', '16px', '32px', '100px'],
  borders: [0, `1px solid`, `4px solid`],
  borderStyles: ['solid'],
  borderWidths: ['0', '1px', '2px', '4px'],
  breakpoints: ['40em', '52em', '64em'],
  shadows: [
    '0',
    '0 0.7px 2.2px -8px rgba(0, 0, 0, 0.18),0 1.7px 5.4px  rgba(0, 0, 0, 0.129),0 3.1px 10.1px rgba(0, 0, 0, 0.107),0 5.6px 18.1px rgba(0, 0, 0, 0.09),0 10.4px 33.8px rgba(0, 0, 0, 0.073),0 25px 81px rgba(0, 0, 0, 0.051)',
    'rgba(0, 0, 0, 0.10) 0px 0.7px 2.2px -8px, rgba(0, 0, 0, 0.04) 0px 1.7px 2.4px, rgba(0, 0, 0, 0.106) 0px 3.1px 8.1px, rgba(0, 0, 0, 0.04) 0px 5.6px 12.1px, rgba(0, 0, 0, 0.045) 0px 4.4px 4.8px, rgba(0, 0, 0, 0.05) 0px 15px 41px'
  ],
  opacity: {
    disabled: 0.4
  },
  zIndices: [0, 9, 99, 999, 9999]
}

export default theme

/* EVERYTHING BELOW HERE IS THEMEV2 */

// safe to change
const deviceWidthPxs = {
  phone: 480,
  tablet: 868,
  desktop: 1124
}

const spaces = {
  phone: {
    sm: '',
    small: '',
    default: '8px',
    md: '8px',
    medium: '8px',
    lg: '20px',
    large: '20px'
  },
  default: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  },
  tablet: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  },
  desktop: {
    sm: '',
    small: '',
    default: '10px',
    md: '10px',
    medium: '10px',
    lg: '20px',
    large: '20px'
  }
}

const fontSizes = {
  phone: {
    sm: '13px',
    small: '13px',
    default: '16px',
    md: '18px',
    medium: '18px',
    lg: '20px',
    large: '20px'
  },
  default: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  },
  tablet: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  },
  desktop: {
    sm: '13px',
    small: '13px',
    default: '18px',
    md: '26px',
    medium: '26px',
    lg: '28px',
    large: '28px'
  }
}

export const maxWidth = '1450px'

// dont change
export const devices = {
  phone: `${deviceWidthPxs.phone}px`,
  tablet: `${deviceWidthPxs.tablet}px`,
  desktop: `${deviceWidthPxs.desktop}px`
}

// these are
// one of size = sm, default, md, lg
// one of device = phone, tablet, desktop
export function space(size = 'default', device = 'default') {
  return spaces[device][size]
}

export function fontSize(size = 'default', device = 'default') {
  return fontSizes[device][size]
}
