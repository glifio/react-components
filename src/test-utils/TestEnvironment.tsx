import { ReactNode } from 'react'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { CoinType } from '@glif/filecoin-address'
import LotusRPCEngine from '@glif/filecoin-rpc-client'
import { Environment, Network, networks } from '../services/EnvironmentProvider'

const OptionalApolloMockProvider = ({ children, mocks, withApollo }) => {
  if (withApollo || mocks.length > 0)
    return <MockedProvider mocks={mocks}>{children}</MockedProvider>

  return <>{children}</>
}

export const TestEnvironment = ({
  children,
  withApollo,
  apolloMocks
}: TestEnvironmentProps) => (
  <Environment
    coinType={CoinType.TEST}
    networkName={Network.CALIBRATION}
    nodeStatusUrl={networks[Network.CALIBRATION].nodeStatusUrl}
    graphUrl={networks[Network.CALIBRATION].graphUrl}
    graphSecure={networks[Network.CALIBRATION].graphSecure}
    lotusApiUrl={networks[Network.CALIBRATION].lotusApiUrl}
    explorerUrl='https://explorer-calibration.glif.link'
    lotusApi={
      new LotusRPCEngine({
        apiAddress: networks[Network.CALIBRATION].lotusApiUrl
      })
    }
  >
    <OptionalApolloMockProvider withApollo={withApollo} mocks={apolloMocks}>
      {children}
    </OptionalApolloMockProvider>
  </Environment>
)

type TestEnvironmentProps = {
  children: ReactNode
  withApollo?: boolean
  apolloMocks?: MockedResponse[]
}

TestEnvironment.defaultProps = {
  withApollo: false,
  apolloMocks: []
}
