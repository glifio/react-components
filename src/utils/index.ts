export { default as isMobileOrTablet } from './isMobileOrTablet'
export { default as useDesktopBrowser } from './useDesktopBrowser'
export { default as useChromeDesktopBrowser } from './useChromeDesktopBrowser'
export * from './useGetMessage'
export * from './useStateReadState'
export * from './actorCode'
export * from './isAddrEqual'
