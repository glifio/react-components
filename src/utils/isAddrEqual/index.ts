import { Address } from '../../generated/graphql'
import convertAddrToPrefix from '../convertAddrToPrefix'

export const isAddrEqual = (
  address: Address,
  address2: Address | string
): boolean => {
  if (!address || !address2) return false

  if (typeof address2 === 'string') {
    return (
      convertAddrToPrefix(address2) === convertAddrToPrefix(address.id) ||
      convertAddrToPrefix(address2) === convertAddrToPrefix(address.robust)
    )
  }

  return (
    convertAddrToPrefix(address2.id) === convertAddrToPrefix(address.id) ||
    convertAddrToPrefix(address2.robust) === convertAddrToPrefix(address.robust)
  )
}
