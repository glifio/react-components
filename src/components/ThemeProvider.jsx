import PropTypes from 'prop-types'
import { ThemeProvider as StyledThemeProvider } from 'styled-components'
import { theme as defaultTheme } from './theme'

const ThemeProvider = ({ children, theme, ...rest }) => {
  return (
    <StyledThemeProvider theme={theme} {...rest}>
      {children}
    </StyledThemeProvider>
  )
}

ThemeProvider.propTypes = {
  theme: PropTypes.object,
  children: PropTypes.node
}

ThemeProvider.defaultProps = {
  theme: defaultTheme,
  children: <></>
}

ThemeProvider.displayName = 'ThemeProvider'

export default ThemeProvider
