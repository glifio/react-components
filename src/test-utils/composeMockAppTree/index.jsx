/* eslint-disable react/prop-types */
import React from 'react'
import { mockWalletProviderInstance } from '../../../__mocks__/@glif/filecoin-wallet-provider'
import WalletProviderWrapper from '../../services/WalletProvider'
import { initialState as walletProviderInitialState } from '../../services/WalletProvider/state'
import { composeWalletProviderState } from './composeState'
import { TestEnvironment } from '../TestEnvironment'

jest.mock('../../services/WalletProvider')

const Index = (statePreset = 'preOnboard', options = {}) => {
  // here you can pass a walletProviderInitialState and a preset to shape the store how you want it for testing
  const initialState = composeWalletProviderState(
    options?.walletProviderInitialState || walletProviderInitialState,
    statePreset
  )

  let walletProviderCache = { ...initialState }

  const cacheWalletProviderState = state => {
    walletProviderCache = { ...state }
    return <></>
  }

  const getWalletProviderState = () => walletProviderCache

  const Tree = ({ children }) => {
    return (
      <TestEnvironment>
        <WalletProviderWrapper
          options={options}
          statePreset={statePreset}
          getState={cacheWalletProviderState}
          initialState={initialState}
        >
          {children}
        </WalletProviderWrapper>
      </TestEnvironment>
    )
  }

  return {
    Tree,
    walletProvider: mockWalletProviderInstance,
    getWalletProviderState
  }
}

/**
 * This function is a wrapper that mocks everything the filecoin app needs for testing
 *
 * VALID OPTIONS:
 * mockConverterInstance (an object that stubs the converter functionality)
 * walletProviderReducer (a reducer to stub the state handling of the wallet provider)
 * walletProviderInitialState(a reducer to stub the state of the walletProvider context)
 * walletProviderDispatch(a dispatcher to mock/read calls to the walletProvider's state)
 * pathname
 * query
 */

export default Index
