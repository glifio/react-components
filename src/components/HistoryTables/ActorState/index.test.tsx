import React from 'react'
import {
  render,
  act,
  fireEvent,
  RenderResult,
  getByText,
  waitFor
} from '@testing-library/react'

import {
  MULTISIG_ACTOR_ADDRESS,
  MULTISIG_ACTOR_ID
} from '../../../test-utils/constants'
import { TestEnvironment } from '../../../test-utils/TestEnvironment'
import { isAddrEqual } from '../../../utils/isAddrEqual'
import ThemeProvider from '../../ThemeProvider'
import theme from '../../theme'
import { ActorState } from '.'

jest
  .spyOn(require('@glif/filecoin-rpc-client'), 'default')
  .mockImplementation(() => {
    return {
      request: jest.fn(async (method, address) => {
        if (!isAddrEqual(address, MULTISIG_ACTOR_ADDRESS))
          throw new Error('getting actor: actor not found')

        switch (method) {
          case 'StateReadState':
            return {
              Balance: '1000000000000000000',
              Code: {
                '/': 'bafk2bzacec66wmb4kohuzvuxsulhcgiwju7sqkldwfpmmgw7dbbwgm5l2574q'
              },
              State: {
                Signers: ['t014278'],
                NumApprovalsThreshold: 1,
                NextTxnID: 0,
                InitialBalance: '1000000000000000000',
                StartEpoch: 239252,
                UnlockDuration: 100,
                PendingTxns: {
                  '/': 'bafy2bzaceamp42wmmgr2g2ymg46euououzfyck7szknvfacqscohrvaikwfay'
                }
              }
            }
          case 'MsigGetAvailableBalance':
            return '1000000000000000000'
          default:
            throw new Error('unknown method')
        }
      })
    }
  })

jest
  .spyOn(require('../../../generated/graphql'), 'useAddressQuery')
  .mockImplementation(() => ({
    data: {
      address: {
        id: MULTISIG_ACTOR_ID,
        robust: MULTISIG_ACTOR_ADDRESS
      }
    },
    loading: false,
    error: null
  }))

describe('ActorState', () => {
  test('it renders the loading view correctly', async () => {
    let result: RenderResult | null = null
    await act(async () => {
      result = render(
        <TestEnvironment withApollo>
          <ThemeProvider theme={theme}>
            <ActorState address={MULTISIG_ACTOR_ADDRESS} />
          </ThemeProvider>
        </TestEnvironment>
      )

      expect(
        getByText(result.container, 'Locating this actor on the blockchain...')
      ).toBeInTheDocument()
    })
  })

  test('it renders the not found state correctly', async () => {
    let result: RenderResult | null = null
    await act(async () => {
      result = render(
        <TestEnvironment withApollo>
          <ThemeProvider theme={theme}>
            <ActorState address='t011' />
          </ThemeProvider>
        </TestEnvironment>
      )
    })

    expect(
      getByText(result!.container, 'Actor not found: t011')
    ).toBeInTheDocument()
  })

  test('it renders the actor state correctly', async () => {
    let result: RenderResult | null = null
    await act(async () => {
      result = render(
        <TestEnvironment withApollo>
          <ThemeProvider theme={theme}>
            <ActorState address={MULTISIG_ACTOR_ADDRESS} />
          </ThemeProvider>
        </TestEnvironment>
      )

      await waitFor(() => {
        expect(
          getByText(result!.container, 'Click to show actor state ↓')
        ).toBeInTheDocument()
      })

      fireEvent.click(
        getByText(result!.container, 'Click to show actor state ↓')
      )
    })

    expect(getByText(result!.container, 'InitialBalance')).toBeInTheDocument()
    expect(getByText(result!.container, 'NextTxnID')).toBeInTheDocument()
    expect(
      getByText(result!.container, 'NumApprovalsThreshold')
    ).toBeInTheDocument()
    expect(
      getByText(result!.container, 'PendingTxns (CID)')
    ).toBeInTheDocument()
    expect(getByText(result!.container, 'Signers')).toBeInTheDocument()
    expect(getByText(result!.container, 'StartEpoch')).toBeInTheDocument()
    expect(getByText(result!.container, 'UnlockDuration')).toBeInTheDocument()

    expect(result!.container.firstChild).toMatchSnapshot()
  })
})
