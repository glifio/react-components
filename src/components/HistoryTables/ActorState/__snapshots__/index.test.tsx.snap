// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActorState it renders the actor state correctly 1`] = `
.c8 {
  vertical-align: middle;
}

.c5 h4 {
  margin: 0;
  color: var(--gray-dark);
}

.c5 p {
  margin: 0;
  line-height: 1.5;
}

.c7 {
  position: relative;
  cursor: pointer;
}

.c7 svg {
  -webkit-transition: -webkit-transform 0.1s ease-out;
  -webkit-transition: transform 0.1s ease-out;
  transition: transform 0.1s ease-out;
}

.c7 .copy-text {
  position: absolute;
  top: calc(50% - 0.2em);
  left: calc(100%);
  padding-left: 0.5em;
  color: var(--purple-medium);
  opacity: 0;
  -webkit-transform: translateY(-50%);
  -ms-transform: translateY(-50%);
  transform: translateY(-50%);
  -webkit-transition: opacity 0.2s ease-in;
  transition: opacity 0.2s ease-in;
  pointer-events: none;
}

.c7:hover svg {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
}

.c7:hover .copy-text {
  opacity: 1;
  pointer-events: all;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 0.35em;
  line-height: 1.5;
}

.c6 > a {
  color: var(--purple-medium);
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6 > a:hover {
  color: var(--purple-medium);
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c6 > a svg {
  -webkit-transition: 0.24s ease-in-out;
  transition: 0.24s ease-in-out;
}

.c6 > a svg:hover {
  -webkit-transform: scale(1.25);
  -ms-transform: scale(1.25);
  transform: scale(1.25);
  fill: var(--purple-medium);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: var(--space-m);
}

.c2 > p,
.c2 > hr {
  margin: 0.5em 0;
}

.c2 > p[role='button'] {
  margin: 0;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1em;
  line-height: 2.5em;
  text-align: left;
  padding-left: 0em;
}

.c3 > div:first-child {
  -webkit-flex: 0 1 25%;
  -ms-flex: 0 1 25%;
  flex: 0 1 25%;
  min-width: 200px;
  text-transform: capitalize;
}

.c3 > div:last-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.75em;
  word-break: break-word;
}

.c3 > div:last-child > p[role='button'] {
  margin: 0;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1em;
  line-height: 2.5em;
  text-align: left;
  padding-left: 1.5em;
}

.c4 > div:first-child {
  -webkit-flex: 0 1 25%;
  -ms-flex: 0 1 25%;
  flex: 0 1 25%;
  min-width: 200px;
  text-transform: capitalize;
}

.c4 > div:last-child {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.75em;
  word-break: break-word;
}

.c4 > div:last-child > p[role='button'] {
  margin: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  gap: var(--space-l);
  margin: 1em 0;
  font-size: var(--font-size-xl);
}

.c0 .side-bar {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: var(--space-m);
  font-size: 1rem;
}

.c1 {
  margin: 0;
  color: var(--purple-medium);
}

<div>
  <div
    class="c0"
  >
    <h2
      class="c1"
    >
      Actor Overview
    </h2>
  </div>
  <hr />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <div>
        Robust address
      </div>
      <div>
        f2m4f2dv7m35skytoqzsyrh5wqz3kxxfflxsha5za
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        ID address
      </div>
      <div>
        t0201
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Actor name
      </div>
      <div>
        Multisig
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Actor code
      </div>
      <div>
        bafk2bzacec66wmb4kohuzvuxsulhcgiwju7sqkldwfpmmgw7dbbwgm5l2574q
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Balance
      </div>
      <div>
        1
         FIL
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        Available Balance
      </div>
      <div>
        1
         FIL
      </div>
    </div>
    <div
      class="c3"
    >
      <div>
        State
      </div>
      <div>
        <p
          role="button"
        >
          Click to hide actor state â†‘
        </p>
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        Signers
      </div>
      <div>
        <div
          class="c5"
        >
          <div
            class="c6"
            color="var(--purple-medium)"
          >
            <a
              href="https://explorer-calibration.glif.link/actor/?address=f2m4f2dv7m35skytoqzsyrh5wqz3kxxfflxsha5za"
              rel="noreferrer noopener"
              target="_blank"
            >
              f2m4f2 ... sha5za
            </a>
            <span
              class="c7"
              color="var(--purple-medium)"
              role="button"
            >
              <svg
                class="c8"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  fill="#1A0066"
                  height="12"
                  width="12"
                  x="4"
                  y="0"
                />
                <rect
                  fill="#E0D7FE"
                  height="12"
                  width="12"
                  x="0"
                  y="4"
                />
              </svg>
              <span
                class="copy-text"
              >
                Copy
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        NumApprovalsThreshold
      </div>
      <div>
        1
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        NextTxnID
      </div>
      <div>
        0
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        InitialBalance
      </div>
      <div>
        6 FIL
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        StartEpoch
      </div>
      <div>
        239252
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        UnlockDuration
      </div>
      <div>
        100
      </div>
    </div>
    <div
      class="c4"
    >
      <div>
        PendingTxns (CID)
      </div>
      <div>
        bafy2bzaceamp42wmmgr2g2ymg46euououzfyck7szknvfacqscohrvaikwfay
      </div>
    </div>
  </div>
</div>
`;
